tests/test_calls.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the correlation ID."""
+import asyncio
+import os
+from functools import partial
+from tempfile import TemporaryDirectory
+
+import pytest
+from ghga_service_commons.api.testing import AsyncTestClient
+from hexkit.providers.akafka.testutils import KafkaFixture, kafka_fixture  # noqa: F401
+
+from pci.adapters.inbound.fastapi_.utils import (
+    InvalidCorrelationIdError,
+    validate_correlation_id,
+)
+from pci.inject import prepare_rest_app
+from pci.models import NonStagedFileRequested
+from tests.fixtures.joint import JointFixture, joint_fixture  # noqa: F401
+from tests.fixtures.utils import PATCH_LOCATION, replacement_middleware
+
+
+@pytest.mark.asyncio
+async def test_rest_call_without_correlation_id(
+    joint_fixture: JointFixture, monkeypatch: pytest.MonkeyPatch  # noqa: F811
+):
+    """Make sure a new ID is generated if a REST api request is received with a missing
+    correlation ID.
+
+    Have to create a new rest client since monkeypatching won't affect the dispatch
+    function of already-instantiated middleware.
+    """
+    monkeypatch.setattr(
```

Comment:
Instead of monkey patching, couldn't you just send test requests with different values in the correlation id header?

Reply:
You're absolutely right 😵‍💫